fileErrors.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2018 The Moov Authors
+// Use of this source code is governed by an Apache License
+// license that can be found in the LICENSE file.
+
+package ach
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/moov-io/base"
+)
+
+// FileTooLongErr is the error given when a file exceeds the maximum possible length
+type FileTooLongErr string
```

Comment:
If we wanted to cut down on the error types what about:

```go
type FileErr struct {
	Description string
}
func (e FileErr) Error() string {
	return fmt.Sprintf("FileErr: %s", e.Description)
}

func NewFileErr(desc string) FileErr {
	return FileErr{desc} 
}
```

Reply:
We could do that, but then we would need to change how we check what kind of error it is. This PR's implementation checks for the same type. We would have to adjust that to additionally check that the details (e.g. `Description` in this case) match. So that basically takes us back to just matching the error strings to do this check, which is more fragile. Of course we could save those strings as constants like we do now, but in that case this whole thing seems kinda redundant...

Maybe instead of using separate types for simple errors (like `FileTooLongErr`), we could do something like this: `var FileTooLongErr error = errors.New("File exceeds maximum possible length")`. I think the `Has` function can be adjusted to deal with this kind of error as well. Then we could just drag out the whole type thing only when we want the error to carry extra information (like `RecordWrongLengthErr`).

Reply:
If Has function can be adjusted this might make everything easier 

Maybe instead of using separate types for simple errors (like FileTooLongErr), we could do something like this: var FileTooLongErr error = errors.New("File exceeds maximum possible length"). I think the Has function can be adjusted to deal with this kind of error as well. Then we could just drag out the whole type thing only when we want the error to carry extra information (like RecordWrongLengthErr).

Reply:
I think `FileTooLongErr := errors.New("...")` is a better way for most error values. Your point on i18n is important too in that having a consolidated place to change in the future will simplify that work.

Reply:
Alright, sounds good! I'll put up a version with this change.